src/jit/lclvars.cpp

```diff
@@ -5576,6 +5620,30 @@ void Compiler::lvaAssignVirtualFrameOffsetsToLocals()
         codeGen->setFramePointerUsed(codeGen->isFramePointerRequired());
     }
 
+#ifdef _TARGET_ARM64_
+    // Decide where to save FP and LR registers. We store FP/LR registers at the bottom of the frame if there is
+    // a frame pointer used (so we get positive offsets from the frame pointer to access locals), but not if we
+    // need a GS cookie AND localloc is used, since we need the GS cookie to protect the saved return value,
+    // and also the saved frame pointer. See CodeGen::genPushCalleeSavedRegisters() for more details about the
+    // frame types. Since saving FP/LR at high addresses is a relatively rare case, force using it during stress.
+    // (It should be legal to use these frame types for every frame).
+
+    if (opts.compJitSaveFpLrWithCalleeSavedRegisters == -1)
```

Comment:
Do we have other cases where we use -1 as a default value? I had an impression that we stored most config values as unsigned and used 0 as default.

What happens if we need the new frame but its using is disabled, will we see the error that this PR fixes? Why do we need such mode?

Reply:
We have a few variables that start as -1. I'm ok changing it to 0==default, 1==disable new types, 2==force using new types.

Yes, if you disable this, you end up with the issue this PR fixes. It is useful for testing. We wouldn't want it in non-DEBUG builds.